### DISCUSSION TITLE
History trend troubleshooting

### DISCUSSION THREAD

**Discussion by StephanusVictor** [riclolsen/json-scada#74] [Q&A] [answered]
Hi Ricardo,

How do I troubleshoot the history trend popup window? Which things to look for? 
The window pops up with the last value plotted, but no trend.
I get "Historian invalid service response!" in the console.  I checked the "hist" table in the postgres db and the data is logged. 

regards
Stephanus

**Comment by riclolsen** [comment, marked as answer]
Hi Stephanus!

Are using the Windows installer? Which version?
I have tested it here and it is working (v0.24). 

Any Postgresql config change? Maybe Windows Firewall is blocking the postgresql port.

The service JSON_SCADA_server_realtime_auth need env variables if defaults are not use to reach Postgresql.

rem Use environment variables to connect (for reading) to PostgreSQL historian (https://www.postgresql.org/docs/current/libpq-envars.html)
nssm set JSON_SCADA_server_realtime_auth AppEnvironmentExtra PGHOSTADDR=127.0.0.1 PGPORT=27017 PGDATABASE=json_scada PGUSER=json_scada PGPASSWORD=json_scada

**Reply by StephanusVictor** [reply]
Thanks Ricardo!
Solved.
It was an authentication issue. I used a different password and wrongfully assumed that the “realtime_service” would detect the “.pgpass” file as the “process_pg_hist” script/service does.

For future readers, you used the mongodb port instead of the postgres one in your answer, it should be  PGPORT=5432

Two additional questions regarding the history + trend.
1.	Would it be possible to easily change the trend time range to more than the default 1 hour. I see it is hard coded in the trend.html but also in the realtime_service?  Some of my processes (water treatment) can be slow and something like 6h would be more useful.
2.	Is it possible to log a tag on an interval in addition to on-change. Maybe I am misunderstanding the description for the “historianPeriod” field, is it for this purpose?  As before, some of my values change slowly and if there is not a history entry within the last hour, the trend return an error (Historian data not found for point *)

(For reference, this is a manual install on an Ubuntu derived distro.)

**Reply by riclolsen** [reply]
Hi Stephanus,

Thanks for spotting the wrong postgresql port!
Nice to know you could install successfully on a ubuntu-derived distro.

1. Yes this is hardcoded, I will consider making it configurable somehow in a future version. Nice suggestion.
2. The historianPeriod parameter was added for this exact purpose but this function was not yet implemented, currently only changes are saved to the database. This should be also implemented on a future release. You can try replicating values periodically with a bit of postgresql programming.

I am sorry the development of the project is very slow now because of lack of time.

Best regards,
Ricardo
